src/components/Calculator.js

```diff
@@ -1,28 +1,92 @@
 /* eslint-disable react/destructuring-assignment */
 import React from 'react';
 import PropTypes from 'prop-types'; /* eslint-disable react/forbid-prop-types */
-import Display from './Display';
-import Keypad from './Keypad';
-import Operations from './Operations';
+import calculate from '../logic/calculate';
+
 import './Calculator.css';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: '',
+      next: '',
+      operation: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  componentDidUpdate() {
+    const { total, next } = this.state;
+    const display = document.querySelector('.display');
+    if (next != null) {
+      // @ts-ignore
+      display.value = next;
+    } else if (total != null) {
+      // @ts-ignore
+      display.value = total;
+    }
   }
 
+  handleClick = (e) => {
+    if (e.target.textContent === 'AC') {
+      const display = document.querySelector('.display');
+      // @ts-ignore
+      display.value = 0;
+    }
+    const { total, next, operation } = this.state;
+    this.setState(calculate({ total, next, operation }, e.target.textContent));
+  };
+
   render() {
     return (
       <div className="calculator">
-        <Display />
-        <Keypad digits={this.props.nums} />
-        <Operations ops={this.props.ops} />
+        <div id="display-result" className="display-result">
+          <input className="display" placeholder="0" />
+        </div>
+
+        <div className="numbers-container">
+          <button
+            type="button"
+            className="ac double-width light-grey"
+            onClick={this.handleClick}
```

Comment:
Please add the '%' button as well and make sure that the % button works as modulus.

![image](https://user-images.githubusercontent.com/38088818/170507212-d5de0246-9b5f-472e-8910-c8f5d1f40f66.png)